proto_defs/internal/ledger.proto

```diff
@@ -0,0 +1,30 @@
+syntax = "proto3";
+
+package particle.firmware;
+
+import "cloud/ledger.proto";
+import "nanopb.proto";
+
+/**
+ * Ledger info.
+ */
+message LedgerInfo {
```

Comment:
Is there a need for a sequence number in the unlikely event of out-of-order Ledger sync because of slow transfer networks and modem quirks?

Reply:
There's `update_count` that gets incremented every time the ledger is modified but it's only used internally by Device OS so that it knows exactly which snapshot of ledger data exactly is being synchronized at a given time. It was introduced to work around a subtle race condition around [this code](https://github.com/particle-iot/device-os/blob/59325889b628f591756c40c9cc5fa33dc73c05a9/system/src/ledger/ledger.cpp#L878-L882) possible when a device-to-cloud ledger is modified by the user while it's already being synchronized with the cloud. Without the counter, the ledger manager may prematurely clear the persistent `syncPending` flag and if the device happens to reset around the same time, the last change to the ledger wouldn't be synchronized when the device restarts until the ledger is modified again.

Out-of-order syncs are not possible because only one sync can be in progress at a time. In the worst case, the system may synchronize the same ledger data twice after recovering from a disconnect that happened at the end of the previous sync attempt.